pkg/audit/manager.go

```diff
@@ -226,17 +236,11 @@ func (am *Manager) auditResources(ctx context.Context) ([]*constraintTypes.Resul
 			}
 
 			for _, obj := range objList.Items {
-				ns := &corev1.Namespace{}
-				if obj.GetNamespace() != "" {
-					if err := am.client.Get(ctx, types.NamespacedName{Name: obj.GetNamespace()}, ns); err != nil {
-						am.log.Error(err, "Unable to look up object namespace", "group", gv.Group, "version", gv.Version, "kind", kind)
-						continue
-					}
-				}
+				objNs := nsCache[obj.GetNamespace()]
```

Comment:
One potential race condition: a namespace/resource gets created mid-audit and therefore is not in the initial list. I think that's the value of the lazy cache.

Apologies for not noticing it earlier.